source/opt/const_folding_rules.cpp

```diff
@@ -376,13 +376,7 @@ ConstantFoldingRule FoldVectorTimesMatrix() {
     assert(c1->type()->AsVector()->element_type() == element_type &&
            c2->type()->AsMatrix()->element_type() == vector_type);
 
-    // Get a float vector that is the result of vector-times-matrix.
-    std::vector<const analysis::Constant*> c1_components =
-        c1->GetVectorComponents(const_mgr);
-    std::vector<const analysis::Constant*> c2_components =
-        c2->AsMatrixConstant()->GetComponents();
     uint32_t resultVectorSize = result_type->AsVector()->element_count();
-
     std::vector<uint32_t> ids;
 
     if ((c1 && c1->IsZero()) || (c2 && c2->IsZero())) {
```

Comment:
This code is exactly the same as the one in the other function.
Same for the whole logic, except one case it reads from c1, and the other from c2.
Isn't there some refactorization we can do?

Reply:
One function does a vector times matrix and the other does a matrix times vector. We cannot just flip the two. 


v x M = M^t x v

We would have to do a transpose as well. However, there could be another bug in the code. Should we be checking if the matrices are row or column major? In any case, I view this as a follow up issue. It will hide the actual fix.